styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -0,0 +1,288 @@
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.entity.EntityMiddleware;
+import com.spotify.apollo.entity.JacksonEntityCodec;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.Middleware;
+import com.spotify.apollo.route.Route;
+import com.spotify.styx.api.cli.RunStateDataPayload;
+import com.spotify.styx.api.cli.RunStateDataPayload.RunStateData;
+import com.spotify.styx.model.Backfill;
+import com.spotify.styx.model.BackfillBuilder;
+import com.spotify.styx.model.BackfillInput;
+import com.spotify.styx.model.Partitioning;
+import com.spotify.styx.model.Workflow;
+import com.spotify.styx.model.WorkflowId;
+import com.spotify.styx.model.WorkflowInstance;
+import com.spotify.styx.serialization.Json;
+import com.spotify.styx.state.RunState;
+import com.spotify.styx.state.StateData;
+import com.spotify.styx.storage.Storage;
+import com.spotify.styx.util.RandomGenerator;
+import com.spotify.styx.util.ReplayEvents;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import okio.ByteString;
+
+public final class BackfillResource {
+
+  static final String BASE = "/backfills";
+
+  private final Storage storage;
+
+  public BackfillResource(Storage storage) {
+    this.storage = Objects.requireNonNull(storage);
+  }
+
+  public Stream<? extends Route<? extends AsyncHandler<? extends Response<ByteString>>>> routes() {
+    final EntityMiddleware em =
+        EntityMiddleware.forCodec(JacksonEntityCodec.forMapper(Json.OBJECT_MAPPER));
+
+    final List<Route<AsyncHandler<Response<ByteString>>>> routes = Stream.of(
+        Route.with(
+            em.serializerDirect(BackfillsPayload.class),
+            "GET", BASE,
+            this::getBackfills),
+        Route.with(
+            em.response(BackfillInput.class, Backfill.class),
+            "POST", BASE,
+            rc -> this::postBackfill),
+        Route.with(
+            em.serializerResponse(BackfillPayload.class),
+            "GET", BASE + "/<bid>",
+            rc -> getBackfill(arg("bid", rc))),
+        Route.with(
+            em.serializerResponse(Void.class),
+            "DELETE", BASE + "/<bid>",
+            rc -> haltBackfill(arg("bid", rc))),
+        Route.with(
+            em.response(Backfill.class),
+            "PUT", BASE + "/<bid>",
+            rc -> payload -> updateBackfill(arg("bid", rc), payload))
+    )
+        .map(r -> r.withMiddleware(Middleware::syncToAsync))
+        .collect(toList());
+
+    return Api.prefixRoutes(routes, V1);
+  }
+
+  private BackfillsPayload getBackfills(RequestContext requestContext) {
+    final Optional<String> componentOpt = requestContext.request().parameter("component");
+    final Optional<String> statusesFlagOpt = requestContext.request().parameter("status");
```

Comment:
this flag could be more explicit, `include_status` or something, just the word `status` doesn't really say enough: is it filtering, inclusion of status, what values are possible for the parameter, would `?status=false` or `?status=0` disable the statuses?